manifests/init.pp

```diff
@@ -12,18 +12,31 @@
 #     /mnt/mount:
 #       src: "nfs-server.local:/export/path"
 #       fstype: "nfs4"
-#       opts: "defaults,nosuid,ro"
+#       opts: "defaults,nosuid,nodev,ro"
 # ```
 #
+# @param required_pkgs
+#   Packages that need to be installed for NFS mounts to work.
+# 
 # @example
 #   include profile_nfs_client
 class profile_nfs_client (
+  Array[ String ] $masked_services,
+  Hash            $mountmap,
   Array[ String ] $required_pkgs,
-  Hash $mountmap,
 ) {
 
   ensure_packages( $required_pkgs )
 
+  each($masked_services) | $service |
```

Comment:
`unit` is the correct systemctl work to use instead of `service`.

I'll switch this up to use `$masked_units`, etc.